test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -375,31 +378,60 @@ static void test_out_in_common(ipsec_test_flags *flags,
 		},
 	};
 
+	test.flags = *flags;
+
+	if (flags->stats == IPSEC_TEST_STATS_PROTO_ERR)
+		test.in[0].status.error.proto = 1;
+
+	if (flags->stats == IPSEC_TEST_STATS_SUCCESS) {
+		CU_ASSERT_EQUAL(odp_ipsec_stats(sa_out, &stats), 0);
+		CU_ASSERT_EQUAL(stats.success, 0);
+		CU_ASSERT_EQUAL(odp_ipsec_stats(sa_in, &stats), 0);
+		CU_ASSERT_EQUAL(stats.success, 0);
+	}
+
 	ipsec_check_out_in_one(&test, sa_out, sa_in);
 
+	if (flags->stats == IPSEC_TEST_STATS_SUCCESS) {
+		CU_ASSERT_EQUAL(odp_ipsec_stats(sa_out, &stats), 0);
+		CU_ASSERT_EQUAL(stats.success, 1);
+		CU_ASSERT_EQUAL(odp_ipsec_stats(sa_in, &stats), 0);
+		CU_ASSERT_EQUAL(stats.success, 1);
+	}
+
+	if (flags->stats == IPSEC_TEST_STATS_PROTO_ERR) {
+		CU_ASSERT_EQUAL(odp_ipsec_stats(sa_in, &stats), 0);
+		CU_ASSERT_EQUAL(stats.proto_err, 1);
+	}
```

Comment:
Here the success case could test that error counters are zero and the error case that the success counters are zero,

Reply:
Agreed. Fixed in v7